scan_rr.go

```diff
@@ -1321,8 +1323,10 @@ func setNSEC3PARAM(h RR_Header, c chan lex, o, f string) (RR, *ParseError, strin
 	rr.Iterations = uint16(i)
 	<-c
 	l = <-c
-	rr.SaltLength = uint8(len(l.token))
-	rr.Salt = l.token
+	if l.token != "-" {
+		rr.SaltLength = uint8(len(l.token))
```

Comment:
As an aside @miekg, is there a reason NSEC3.SaltLength is the byte length and NSEC3Param.SaltLength is the hex length? Is that a bug?

Reply:
hmm, looks like it, those fields are identical to rfc 5155